debezium-server/debezium-server-eventhubs/src/main/java/io/debezium/server/eventhubs/EventHubsChangeConsumer.java

```diff
@@ -0,0 +1,182 @@
+                    customProducer.get().getFullyQualifiedNamespace());
+            return;
+        }
+
+        final Config config = ConfigProvider.getConfig();
+        connectionString = config.getValue(PROP_CONNECTION_STRING_NAME, String.class);
+        eventHubName = config.getValue(PROP_EVENTHUB_NAME, String.class);
+
+        // optional config
+        partitionID = config.getOptionalValue(PROP_PARTITION_ID, String.class).orElse("");
+        partitionKey = config.getOptionalValue(PROP_PARTITION_KEY, String.class).orElse("");
+        maxBatchSize = config.getOptionalValue(PROP_MAX_BATCH_SIZE, Integer.class).orElse(0);
+
+        String finalConnectionString = String.format(CONNECTION_STRING_FORMAT, connectionString, eventHubName);
+
+        try {
+            producer = new EventHubClientBuilder().connectionString(finalConnectionString).buildProducerClient();
+        }
+        catch (Exception e) {
+            throw new DebeziumException(e);
+        }
+
+        LOGGER.info("Using default Event Hubs client for namespace '{}'", producer.getFullyQualifiedNamespace());
+    }
+
+    @PreDestroy
+    void close() {
+        try {
+            producer.close();
+            LOGGER.info("Closed Event Hubs producer client");
+        }
+        catch (Exception e) {
+            LOGGER.warn("Exception while closing Event Hubs producer: {}", e);
+        }
+    }
+
+    @Override
+    public void handleBatch(List<ChangeEvent<Object, Object>> records,
+                            RecordCommitter<ChangeEvent<Object, Object>> committer)
+            throws InterruptedException {
+        LOGGER.trace("Event Hubs sink adapter processing change events");
+
+        CreateBatchOptions op = new CreateBatchOptions().setPartitionId(partitionID);
+        if (partitionKey != "") {
+            op.setPartitionKey(partitionKey);
+        }
+        if (maxBatchSize.intValue() != 0) {
+            op.setMaximumSizeInBytes(maxBatchSize);
+        }
+
+        EventDataBatch batch = producer.createBatch(op);
+
+        // this loop just adds records to the batch
+        for (ChangeEvent<Object, Object> record : records) {
+            LOGGER.trace("Received event '{}'", record);
+            if (null == record.value()) {
+                continue;
+            }
+            EventData eventData = null;
+
+            if (record.value() instanceof String) {
+                eventData = new EventData((String) record.value());
+            }
+            else if (record.value() instanceof byte[]) {
+                eventData = new EventData(getBytes(record.value()));
+            }
+            try {
+                if (!batch.tryAdd(eventData)) {
+                    LOGGER.warn("Event data was too large to fit in the batch - {}", record);
+                }
+            }
+            catch (IllegalArgumentException e) {
+                LOGGER.warn("Event data was null - {}", e.getMessage());
+            }
+            catch (AmqpException e) {
```

Comment:
AMQP exception? Can this actually happen here?

in general, the swalled exceptions here look a bit concerning. They'd result in missed events, I think we should rather re-throw them, then e.g. giving the user a chance to reconfigure their maximum message size before retrying.

Reply:
yes, the AMQP exception is thrown by `tryAdd`. Thanks for pointing out the error handling in general. I have updated relevant parts to throw `DebeziumException`